ui/src/components/OrderHistory.jsx

```diff
@@ -1,117 +1,30 @@
+import { ORDERS } from "../config";
+import useMessage from "../hooks/messages";
 
 export default function OrderHistory() {
-  const orders = [
-    {
-      sender: {
-        name: "qwer",
-        surname: "asdf",
-        number: "+6546464646",
-        city: "Vilnius",
-        address: "asdfafds g. 10",
-      },
-      recipient: {
-        name: "dsfgsrg",
-        surname: "dfghdfgdh",
-        number: "+98494984484",
-        city: "Vilnius",
-        address: "sdfdsfsf g. 10",
-      },
-      email: "sdfgsgfd@dfgdfg.dsf",
-      pickUpDate: new Date(2021, 3, 28, 11, 0),
-      status: "New",
-      packageId: 1,
-    },
-    {
-      sender: {
-        name: "fghnfhgn",
-        surname: "fnhgn",
-        number: "+6546464646",
-        city: "Vilnius",
-        address: "asdfafds g. 10",
-      },
-      recipient: {
-        name: "bghdn",
-        surname: "dfbdfgb",
-        number: "+98494984484",
-        city: "Vilnius",
-        address: "sdfdsfsf g. 10",
-      },
-      email: "sfaf@gmail.com",
-      pickUpDate: new Date(2021, 3, 10, 18, 50),
-      status: "New",
-      packageId: 2,
-    },
-    {
-      sender: {
-        name: "nhtrh",
-        surname: "fbggbd",
-        number: "+6546464646",
-        city: "Vilnius",
-        address: "vfvgv g. 10",
-      },
-      recipient: {
-        name: "dsfgsrg",
-        surname: "bbvgb",
-        number: "+98494984484",
-        city: "Vilnius",
-        address: "sdfdsfsf g. 10",
-      },
-      email: "asdfsd@gmail.com",
-      pickUpDate: new Date(2021, 4, 7, 9, 5),
-      status: "Delivered",
-      packageId: 5,
-    },
-  ];
-  const packageSizes = [
-    {
-      id: 1,
-      type: "Box",
-      name: "Small Box",
-      width: 35,
-      height: 16,
-      length: 45,
-      weight: 2,
-    },
-    {
-      id: 2,
-      type: "Box",
-      name: "Medium Box",
-      width: 46,
-      height: 46,
-      length: 64,
-      weight: 20,
-    },
-    {
-      id: 3,
-      type: "Box",
-      name: "Large Box",
-      width: 150,
-      height: 150,
-      length: 150,
-      weight: 30,
-    },
-    {
-      id: 4,
-      type: "Document",
-      name: "Small Letter",
-      weight: 100,
-      length: 24,
-      width: 16,
-    },
-    {
-      id: 5,
-      type: "Document",
-      name: "Large Letter",
-      weight: 750,
-      length: 35,
-      width: 25,
-    },
-  ];
+  const [orders, setOrders] = useState([]);
+  const { displayError } = useMessage();
+  useEffect(() => {
+    (async function fetchData() {
+      try {
+        const response = await fetch(ORDERS);
+        let responseJson = await response.json();
+        for (let index in responseJson.data) {
+          responseJson.data[index].pickupDateTime = new Date(
+            responseJson.data[index].pickupDateTime
+          );
+        }
+        setOrders(responseJson.data);
+      } catch (e) {
+        displayError("Failed to load orders.");
+      }
+    })();
+  }, []);
 
```

Comment:
displayError as dependency is needed here

Reply:
it seems if I add the missing dependency, the API gets called non-stop

Reply:
added displayError ref to avoid this warning